soc/silabs/Kconfig

```diff
@@ -2,10 +2,10 @@
 # Copyright (c) 2018 Gil Benkoe
 # SPDX-License-Identifier: Apache-2.0
 
-if SOC_FAMILY_SILABS_S0 || SOC_FAMILY_SILABS_S1 || SOC_FAMILY_SILABS_S2
-
 rsource "*/Kconfig"
 
+if SOC_FAMILY_SILABS_S0 || SOC_FAMILY_SILABS_S1 || SOC_FAMILY_SILABS_S2
+
```

Comment:
no

Reply:
why has this been broken again?

Reply:
Because of @jerome-pouiller's comment here: https://github.com/zephyrproject-rtos/zephyr/pull/76677#discussion_r1713586697

My understanding is that the way it is done now addresses his and your concern.

Reply:
@nordicjm Silabs has 3 (now 4) SoC families and I would like to avoid complex conditions in `Kconfig`. I assume you would like to have the same pattern in every `soc/*/Kconfig` file, that's it?

Maybe the right pattern would be:
```
if SOC_FAMILY_SILABS_S0 || SOC_FAMILY_SILABS_S1 || SOC_FAMILY_SILABS_S2 || SOC_FAMILY_SILABS_SIM3

rsource "*/Kconfig"

if SOC_FAMILY_SILABS_S0 || SOC_FAMILY_SILABS_S1 || SOC_FAMILY_SILABS_S2
...
endif

endif
```

Reply:
No need to change